src/bulk/common.ts

```diff
@@ -899,15 +903,19 @@ export class BulkWriteShimOperation extends AbstractOperation {
     return 'bulkWrite' as const;
   }
 
-  execute(_server: Server, session: ClientSession | undefined): Promise<any> {
+  async execute(
```

Comment:
Will TimeoutContext always be defined in the execute function?

Reply:
Yes, all of our `Operation` classes should now take a non-optional `TimeoutContext` argument in their `execute` functions. This doesn't just accomodate CSOT, but also is how we implement our legacy timeouts for server selection and connection checkout.

If we're not using a CSOT timeout, a timeout will be created in `executeOperation` and passed in, so by the time we hit the execute call, there will always be a `timeoutContext` available. This was already added to `BulkWriteOperation` in a previous PR, but is now plumbed all the way through our Bulk Write implementation to the `BulkWriteShimOperation` here.

Reply:
Oh I see thanks for the clarification!